pages/user/[address]/releases/[releaseId].tsx

```diff
@@ -4,56 +4,54 @@ import { ethers } from "ethers";
 import { useRouter } from "next/router";
 import { useEffect, useState } from "react";
 import { useForm } from "react-hook-form";
-import theFactory from "../../abis/theFactory.json";
-import { Button, ExplorerLink } from "../../components";
+import theFactory from "../../../../abis/theFactory.json";
+import { Button, ExplorerLink } from "../../../../components";
 import {
   dismissNotification,
   errorNotification,
   formatRevert,
   loadingNotification,
-  successNotification
-} from "../../helpers";
-import { useRelease } from "../../queries";
-import { useWalletStore } from "../../stores";
+  successNotification,
+  transformURL
+} from "../../../../helpers";
+import { useRelease } from "../../../../queries";
+import { useWalletStore } from "../../../../stores";
 
 export default function DashboardPage() {
   const { query, push } = useRouter();
-  const { wallet } = useWalletStore();
+  const { wallet, address: connectedAddress } = useWalletStore();
   const [refetchInterval, setRefetchInterval] = useState(0);
   const [address, setAddress] = useState<string>();
   const { status, data, error } = useRelease(address?.toLowerCase(), refetchInterval);
 
   useEffect(() => {
     // @dev The subgraph can take a second to pick up on
     // the blockchain events. This functions polls until there is data.
-    if (query?.address && !data) {
+    if (query?.releaseId && !data) {
       setRefetchInterval(1000);
     } else {
       setRefetchInterval(0);
     }
-  }, [query?.address, data]);
+  }, [query?.releaseId, data]);
 
   const { register, handleSubmit } = useForm();
 
   useEffect(() => {
-    const storedAddress = localStorage.getItem("release_address");
-    if (storedAddress) {
-      setAddress(storedAddress);
-    } else if (query?.address) {
-      setAddress(query.address.toString());
+    if (query?.releaseId) {
+      setAddress(query.releaseId.toString());
     }
-  }, [query?.address]);
+  }, [query?.releaseId]);
 
   if (status === "loading") return <div>Loading release....</div>;
```

Comment:
interesting to understand if you could pull loading out of useQuery so you dont have to compare to a string.

Something like const { status, data, error, loading } = useRelease(address?.toLowerCase(), refetchInterval);
Then the statement could just be 
```loading && return <div>Loading release....</div>;```

This is just an idea not sure if its valid.

Reply:
Yeah I agree. I think it's the `graphql-request` package returning this. A better option is to create a ENUM. I will do that now.

Reply:
can you not just pull it out of the useQuery hook?

Reply:
Yes 🤣 Don't know I didn't do that in the first place. Nice.

Reply:
I think then you could even remove the enum?

Reply:
I did.